project/src/store/action.ts

```diff
@@ -0,0 +1,4 @@
+import { createAction } from '@reduxjs/toolkit';
+import { City } from '../types/offer';
+
+export const changeCity = createAction<{ city: City }>('main/changeCity');
```

Comment:
```suggestion
export const changeCity = createAction<City>('main/changeCity');
```
а почему бы тебе вместо вложенного объекта, просто сразу не передавать город? 🤔